storage/txcache/testutils_test.go

```diff
@@ -119,3 +121,20 @@ func measureWithStopWatch(b *testing.B, function func()) {
 	duration := sw.GetMeasurementsMap()["time"]
 	b.ReportMetric(duration, "time@stopWatch")
 }
+
+// waitTimeout waits for the waitgroup for the specified max timeout.
+// Returns true if waiting timed out.
+// Reference: https://stackoverflow.com/a/32843750/1475331
+func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
```

Comment:
:1st_place_medal: I have used this exact pattern all around the project, never thought of making a function out of it